### PR TITLE
Pull up ListIterable.binarySearch() as default methods.

### PR DESCRIPTION
**Pull request by motlin**

### CODE REVIEW

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/list/ListIterable.java [hunk 1]
```diff
@@ -279,7 +279,10 @@
...
      * @see Collections#binarySearch(List, Object, Comparator)
      */
-    int binarySearch(T key, Comparator<? super T> comparator);
+    default int binarySearch(T key, Comparator<? super T> comparator)
+    {
+        return Collections.binarySearch((List<? extends T>) this, key, comparator);
+    }
 
     /**
      * Searches for the specified object using the binary search algorithm. The elements in this list must implement
```
Comment (motlin) [inline eclipse-collections-api/src/main/java/org/eclipse/collections/api/list/ListIterable.java:288]: Note that this overload `binarySearch(T key)` already casts `this` and is already default. Not sure why we only did one of the two methods, but I'm guessing it had to do with complex generics.

Comment (donraab) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/list/ListIterable.java:288]: Interesting.  Good catch.

**Comment by motlin** [comment]
@nikhilnanivadekar the TeamCity builds are now consistently failing with: `File /mnt/agent/work/1a6b949da01cf326/test-coverage/test-coverage.iml does not exist`

This is failing on master, unrelated to this PR. What are our next steps here?

**Comment by nikhilnanivadekar** [comment]
Can you try reverting the configuration to what it was before? IntellIj Project style instead of Pom. I think that’s the only thing we changed.

**Pull request merged**